cJSON.c

```diff
@@ -693,62 +693,236 @@ void   cJSON_ReplaceItemInArray(cJSON *array,int which,cJSON *newitem)		{cJSON *
+    if(item) item->type = cJSON_NULL;
+    return item;
+}
+cJSON *cJSON_CreateTrue(void)					
+{
+    cJSON *item = cJSON_New_Item();
+    if(item) item->type = cJSON_True;
+    return item;
+}
+
+cJSON *cJSON_CreateFalse(void)					
+{
+    cJSON *item = cJSON_New_Item();
+    if(item) item->type = cJSON_False;
+    return item;
+}
+
+cJSON *cJSON_CreateBool(int b)					
+{
+    cJSON *item = cJSON_New_Item();
+    if(item) item->type = b ? cJSON_True : cJSON_False;
+    return item;
+}
+
+cJSON *cJSON_CreateNumber(double num)			
+{
+    cJSON *item = cJSON_New_Item();
+    if(item)
+    {
+        item->type = cJSON_Number;
+        item->valuedouble = num;
+        item->valueint = (int) num;
+    }
+    return item;
+}
+
+cJSON *cJSON_CreateString(const char *string)	
+{
+    cJSON *item = cJSON_New_Item();
+    if(item)
+    {
+        item->type = cJSON_String;
+        item->valuestring = cJSON_strdup(string);
+        if(!item->valuestring)
+        {
+            cJSON_Delete(item);
+            return 0;
+        }
+    }
+    return item;
+}
+
+cJSON *cJSON_CreateArray(void)					
+{
+    cJSON *item = cJSON_New_Item();
+    if(item) item->type = cJSON_Array;
+    return item;
+}
+
+cJSON *cJSON_CreateObject(void)					
+{
+    cJSON *item = cJSON_New_Item();
+    if(item) item->type = cJSON_Object;
+    return item;
+}
 
 /* Create Arrays: */
-cJSON *cJSON_CreateIntArray(const int *numbers,int count)		{int i;cJSON *n=0,*p=0,*a=cJSON_CreateArray();for(i=0;a && i<count;i++){n=cJSON_CreateNumber(numbers[i]);if(!n){cJSON_Delete(a);return 0;}if(!i)a->child=n;else suffix_object(p,n);p=n;}return a;}
-cJSON *cJSON_CreateFloatArray(const float *numbers,int count)	{int i;cJSON *n=0,*p=0,*a=cJSON_CreateArray();for(i=0;a && i<count;i++){n=cJSON_CreateNumber(numbers[i]);if(!n){cJSON_Delete(a);return 0;}if(!i)a->child=n;else suffix_object(p,n);p=n;}return a;}
-cJSON *cJSON_CreateDoubleArray(const double *numbers,int count)	{int i;cJSON *n=0,*p=0,*a=cJSON_CreateArray();for(i=0;a && i<count;i++){n=cJSON_CreateNumber(numbers[i]);if(!n){cJSON_Delete(a);return 0;}if(!i)a->child=n;else suffix_object(p,n);p=n;}return a;}
-cJSON *cJSON_CreateStringArray(const char **strings,int count)	{int i;cJSON *n=0,*p=0,*a=cJSON_CreateArray();for(i=0;a && i<count;i++){n=cJSON_CreateString(strings[i]);if(!n){cJSON_Delete(a);return 0;}if(!i)a->child=n;else suffix_object(p,n);p=n;}return a;}
+cJSON *cJSON_CreateIntArray(const int *numbers,int count)		
+{
+    int i;
+    cJSON *n = 0, *p = 0, *a = cJSON_CreateArray();
+    for(i=0;a && i<count;i++)
+    {
+        n = cJSON_CreateNumber(numbers[i]);
+        if(!n)
+        {
+            cJSON_Delete(a);
+            return 0;
+        }
+        if(!i)a->child = n;
+        else suffix_object(p,n);
+        p = n;
+    }
+    return a;
+}
+
+cJSON *cJSON_CreateFloatArray(const float *numbers, int count)	
+{
+    int i;
+    cJSON *n = 0, *p = 0, *a = cJSON_CreateArray();
+    for(i=0;a && i<count;i++)
```

Comment:
As you say, but now i have 5 commits in pull request, i thing i should save state and send it to you to check. BTW when there is only 1 thing behind if or loop, i think to get it to curly is overpowered refactoring.